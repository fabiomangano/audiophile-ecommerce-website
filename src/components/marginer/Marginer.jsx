import React from 'react';
import PropTypes from 'prop-types';

const Marginer = ({ direction, margin }) => {
  if (direction === 'vertical')
    return <span style={{ display: 'flex', width: `${margin}rem` }} />;
  return <span style={{ display: 'flex', height: `${margin}rem` }} />;
};

Marginer.defaultProps = {
  direction: 'horizontal',
};

Marginer.propTypes = {
  direction: PropTypes.oneOf(['horizontal', 'vertical']),
  margin: PropTypes.number.isRequired,
};

export default Marginer;
