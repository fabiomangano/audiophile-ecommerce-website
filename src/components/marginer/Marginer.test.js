import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import Marginer from './Marginer';

describe('Marginer', () => {
  it('should render a vertical marginer', () => {
    const tree = create(<Marginer direction="vertical" margin={1} />);
    expect(tree.toJSON()).toMatchSnapshot({
      props: { style: { display: 'flex', width: expect.any(String) } },
    });
  });

  it('should render an horizontal marginer', () => {
    const tree = create(<Marginer direction="horizontal" margin={1} />);
    expect(tree.toJSON()).toMatchSnapshot({
      props: { style: { display: 'flex', height: expect.any(String) } },
    });
  });

  it('should render a horizontal marginer if no direction is provided, with the given margin', () => {
    const wrapper = shallow(<Marginer margin={1} />);
    expect(wrapper.find('span').props('style')).toEqual({
      style: { display: 'flex', height: '1rem' },
    });
  });

  it('should render a horizontal marginer with the given margin', () => {
    const wrapper = shallow(<Marginer direction="horizontal" margin={1} />);
    expect(wrapper.find('span').props('style')).toEqual({
      style: { display: 'flex', height: '1rem' },
    });
  });

  it('should render a vertical marginer with the given margin', () => {
    const wrapper = shallow(<Marginer direction="vertical" margin={1} />);
    expect(wrapper.find('span').props('style')).toEqual({
      style: { display: 'flex', width: '1rem' },
    });
  });
});
