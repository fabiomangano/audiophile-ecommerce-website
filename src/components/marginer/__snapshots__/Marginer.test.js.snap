// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marginer should render a vertical marginer 1`] = `
Object {
  "children": null,
  "props": Object {
    "style": Object {
      "display": "flex",
      "width": Any<String>,
    },
  },
  "type": "span",
}
`;

exports[`Marginer should render an horizontal marginer 1`] = `
Object {
  "children": null,
  "props": Object {
    "style": Object {
      "display": "flex",
      "height": Any<String>,
    },
  },
  "type": "span",
}
`;
